Growth section rewrite

piemets opened the pull request:
I substantially rewritten the Growth section of the handbook and no doubt made a few errors... 

Summary:

- New 'Sales' section which comprises _Sales Operations_ (brand new guide to HubSpot etc.), Demos and Billing. Deleted Trials page as it had become redundant with new pages.
- Renamed 'Content' section to Marketing, put the new _Press_ page under there
- Restructured the sidebar accordingly (pretty sure I messed up the child sections here...) 

This should hopefully give us 4 main sections under Growth which we can then build out with more content, guides etc:
- Community
- Customer Success
- Marketing
- Sales

mariusandra commented:
Hey! This was indeed slightly broken. I fixed what I could find and also moved the files around, do better reflect the sidebar structure in the filesystem. 

In case this is the sidebar you were going for, it now works:

![image](https://user-images.githubusercontent.com/53387/103893150-18e40500-50ed-11eb-9c12-fb0752cf7d89.png)

I'm not qualified to properly review the content though, but it seems fine to me.

piemets commented:
> Hey! This was indeed slightly broken. I fixed what I could find and also moved the files around, do better reflect the sidebar structure in the filesystem.
> 
> In case this is the sidebar you were going for, it now works:
> 
> ![image](https://user-images.githubusercontent.com/53387/103893150-18e40500-50ed-11eb-9c12-fb0752cf7d89.png)
> 
> I'm not qualified to properly review the content though, but it seems fine to me.

Great thank you! Yes that is what I was going for. 

Content-wise, we will next need to then do some work on the Marketing section @jamesefhawkins but I think this is a better way of structuring this information. I'll also do some more writing on Customer Success, but will need to do a bit more work on our support flow first.

jamesefhawkins commented:
There are a few detailed things I think we can improve heree, but this is so much better than _nothing_ that I'm going to merge then we can tweak.
